src/orchestration/orchestration.cpp

```diff
@@ -616,13 +620,95 @@ Ref<OScriptFunction> Orchestration::create_function(const MethodInfo& p_method,
     function->_method = p_method;
     function->_owning_node_id = p_node_id;
     function->_user_defined = p_user_defined;
+    function->_returns_value = MethodUtils::has_return_value(p_method) ;
 
     _functions[p_method.name] = function;
 
     _self->emit_signal("functions_changed");
 
     return function;
 }
+Ref<OScriptFunction> Orchestration::duplicate_function(const StringName& p_name, bool p_include_code)
+{
+    ERR_FAIL_COND_V_MSG(_has_instances(), nullptr, "Cannot duplicate functions, instances exist.");
+    ERR_FAIL_COND_V_MSG(!has_function(p_name), nullptr, "No function exists with the name: " + p_name);
+
+    Ref<OScriptGraph> old_graph = find_graph(p_name);
+    Ref<OScriptFunction> old_function = find_function(p_name);
+    Ref<OScriptFunction> result;
```

Comment:
I might suggest using `new_function` here as the name, to just avoid confusion with the result node